src/NServiceBus.Core/Installation/InstallationSupport.cs

```diff
@@ -40,6 +42,7 @@ static IEnumerable<Type> GetInstallerTypes(FeatureConfigurationContext context)
                 .Where(t => typeof(INeedToInstallSomething).IsAssignableFrom(t) && !(t.IsAbstract || t.IsInterface));
         }
 
+        /// I think we are missing here a concept like a factory which is responsibly to get all installers
```

Comment:
I think here is an explicit concept for installer creation missing which is controlled by the feature itself and can therefore be passed into the ctor of the startup task.